src/Framework/ResultCacheExtension.php

```diff
@@ -18,19 +18,12 @@
 use PHPUnit\Runner\AfterTestFailureHook;
 use PHPUnit\Runner\AfterTestWarningHook;
 use PHPUnit\Runner\BaseTestRunner;
-use PHPUnit\Runner\BeforeFirstTestHook;
-use PHPUnit\Runner\BeforeTestHook;
 use PHPUnit\Runner\TestResultCache;
 
-final class ResultCacheExtension implements BeforeFirstTestHook, BeforeTestHook, AfterSuccessfulTestHook, AfterSkippedTestHook, AfterRiskyTestHook, AfterIncompleteTestHook, AfterTestErrorHook, AfterTestWarningHook, AfterTestFailureHook, AfterLastTestHook
+final class ResultCacheExtension implements AfterSuccessfulTestHook, AfterSkippedTestHook, AfterRiskyTestHook, AfterIncompleteTestHook, AfterTestErrorHook, AfterTestWarningHook, AfterTestFailureHook, AfterLastTestHook
 {
     /**
-     * @var array<string, array> array to keep track of suite/test state and timing information
-     */
-    private $runState = [];
```

Comment:
ping @epdenouden ,
dropping this property will save a lot of memory for huge testbase (like `15023` of mine)